File: ios/ios/Components/ShowItem.swift

import SwiftUI
import TvManiac

struct ShowItem: View {
    var show: ShowUiModel

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            ShowPosterImage(
                imageLoader: ImageLoaderCache.shared.loaderFor(path: show.posterImageUrl, size: .original),
                posterSize: .medium
            )
        }
    }
}

struct ShowItem_Previews: PreviewProvider {
    static var previews: some View {
        ShowItem(show: ShowUiModel(
            id: 1,
            title: "",
            overview: "",
            language: "",
            posterImageUrl: "https://image.tmdb.org/t/p/original/bZGAX8oMDm3Mo5i0ZPKh9G2OcaO.jpg",
            backdropImageUrl: "",
            year: "",
            status: "",
            votes: 21,
            averageVotes: 233.0,
            isInWatchlist: false,
            genreIds: [12, 15]
        ))
    }
}

File: features/search/presenter/src/commonMain/kotlin/com/thomaskioko/tvmaniac/search/presenter/model/ShowItem.kt
package com.thomaskioko.tvmaniac.search.presenter.model

public data class ShowItem(
    val tmdbId: Long = 0,
    val traktId: Long = 0,
    val title: String = "",
    val status: String? = null,
    val voteAverage: Double? = null,
    val year: String? = null,
    val posterImageUrl: String? = null,
    val overview: String? = null,
    val inLibrary: Boolean = false,
)
